geom/src/tessellation.rs

```diff
@@ -0,0 +1,148 @@
+use crate::{Angle, Bounds, Polygon, Pt2D};
+
+// Deliberately not serializable
+/// A tessellated polygon, ready for rendering.
+#[derive(Clone)]
+pub struct Tessellation {
+    /// These points aren't in any meaningful order. It's not generally possible to reconstruct a
+    /// `Polygon` from this.
+    points: Vec<Pt2D>,
+    /// Groups of three indices make up the triangles
+    indices: Vec<u16>,
+}
+
+#[derive(Clone, Debug)]
+pub struct Triangle {
+    pub pt1: Pt2D,
+    pub pt2: Pt2D,
+    pub pt3: Pt2D,
+}
+
+impl From<Polygon> for Tessellation {
+    fn from(polygon: Polygon) -> Self {
+        Self {
+            points: polygon.points,
+            indices: polygon.indices,
+        }
+    }
+}
+
+impl Tessellation {
+    pub fn new(points: Vec<Pt2D>, indices: Vec<usize>) -> Self {
+        Tessellation {
+            points,
+            indices: downsize(indices),
+        }
+    }
+
+    pub fn triangulate(points: Vec<Pt2D>) -> Self {
+        assert!(points.len() >= 3);
+
+        let mut vertices = Vec::new();
+        for pt in &points {
+            vertices.push(pt.x());
+            vertices.push(pt.y());
+        }
+        let indices = downsize(earcutr::earcut(&vertices, &Vec::new(), 2));
+
+        Self { points, indices }
+    }
+
+    /// Returns (points, indices) for rendering
+    pub fn consume(self) -> (Vec<Pt2D>, Vec<u16>) {
+        (self.points, self.indices)
+    }
+
+    pub fn triangles(&self) -> Vec<Triangle> {
+        let mut triangles: Vec<Triangle> = Vec::new();
+        for slice in self.indices.chunks_exact(3) {
+            triangles.push(Triangle {
+                pt1: self.points[slice[0] as usize],
+                pt2: self.points[slice[1] as usize],
+                pt3: self.points[slice[2] as usize],
+            });
+        }
+        triangles
+    }
+
+    pub fn get_bounds(&self) -> Bounds {
```

Comment:
I don't think we need this often enough to cache it and remember to recalculate, like in your branch. Was there a reason for doing that?

Reply:
It was long enough ago that I don't really remember to be honest. I just did some preliminary profiling - and (very anecdotally) it's contributing about 1% of runtime to the `game` bin. Most of this expense though seems to be in the way we are constantly re-recreating the minimap panel.

Anyway, I think it's pretty reasonable to not cache the bounds at this point.